bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/JSONParser.java

```diff
@@ -806,23 +809,21 @@ public State transition(StateMachine sm, char[] buff, int i, int count) throws J
         private void processNonStringValue(ValueType type) throws JsonParserException {
             String str = value();
             if (str.indexOf('.') >= 0) {
+                char ch = str.charAt(0);
                 try {
-                    double doubleValue = Double.parseDouble(str);
-                    switch (type) {
-                        case ARRAY_ELEMENT:
-                            ((ArrayValue) this.currentJsonNode).append(doubleValue);
-                            break;
-                        case FIELD:
-                            ((MapValueImpl<BString, Object>) this.currentJsonNode).put(
-                                    StringUtils.fromString(this.fieldNames.pop()), doubleValue);
-                            break;
-                        case VALUE:
-                            currentJsonNode = doubleValue;
-                            break;
-                        default:
-                            break;
+                    if (ch == '-' && !fromJsonFloatStringMode && !fromJsonDecimalStringMode && isZero(str)) {
```

Comment:
We don't need !fromJsonFloatStringMode in first if condition due to second if else condition fromJsonFloatStringMode.

Reply:
Fixed